src/main/java/org/mskcc/kickoff/upload/jira/FromJiraPmJiraUserRetriever.java

```diff
@@ -41,45 +43,65 @@
     @Value("${jira.rest.path}")
     private String jiraRestPath;
 
-    @Value("${default.pm}")
-    private String defaultPm;
+//    @Value("${default.pm}")
+//    private String defaultPm;
 
-    @Autowired
-    @Qualifier("jiraRestTemplate")
     private RestTemplate restTemplate;
 
+    public FromJiraPmJiraUserRetriever(@Qualifier("jiraRestTemplate") RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
     @Override
     public JiraUser retrieve(String projectManagerIgoName) {
-        projectManagerIgoName = getPmIgoName(projectManagerIgoName);
-
-        LOGGER.info(String.format("Looking for jira user with IGO name \"%s\"", projectManagerIgoName));
+        LOGGER.info(String.format("Looking for jira user with IGO name [%s]", projectManagerIgoName));
 
+        // fetch jira users of pm group
         List<JiraUser> pmJiraUsers = getPmJiraUsers();
+
+        // check if projectManagerIgoName is valid
+        // if valid, then try to match
+        if (isProjectManagerIgoNameValid(projectManagerIgoName)) {
+            Optional<JiraUser> matchedJiraUsers = tryMatchJiraUsers(projectManagerIgoName, pmJiraUsers);
+            if (matchedJiraUsers.isPresent()) return matchedJiraUsers.get();
+        }
+
+        // if not valid or matching failed, use first member
+        if (pmJiraUsers.isEmpty()) {
+            throw new NoPmFoundException(String.format("No Project Manager found in jira group [%s].", pmGroupName));
+        } else {
+            LOGGER.info(String.format("No Project Manager in jira group %s " +
+                    "with IGO name %s: use first member of igo-pm group.", pmGroupName, projectManagerIgoName));
+            // No matching found
+            // set the assignee to the first member of "igo-pm" jira group as returned by the Jira API
+            return pmJiraUsers.get(0);
+        }
+    }
+
+    // invalid cases: null, empty, blank, NA, NO_PM
```

Comment:
This comment can easily get invalid when logic will change and someone won't change it.